thriftbp/ttl_client.go

```diff
@@ -60,6 +73,9 @@ func (c *ttlClient) Close() error {
 		c.state <- state
 	}()
 	state.closed = true
+	if state.cancelRefresh != nil {
+		state.cancelRefresh()
```

Comment:
🔕 (optional) This doesn't join the background goroutine, so it could still be left running / hanging after Close.  (I don't think we guaranteed that before, it's a very useful thing to try to ensure)

Reply:
A possible scenario is that `refresh` is currently blocking on dns, if we try to join here it will block `Close()` for a long time.

And `Close()` calls can often happen in hot path (e.g in a client call, we get a client out of the pool, found out that it's broken, so we `Close` it before opening a new one), which means this can unintentionally block on hot path.